stats/src/test/java/io/airlift/stats/TestGarbageCollectionNotificationInfo.java

```diff
@@ -0,0 +1,50 @@
+package io.airlift.stats;
+
+import com.sun.management.GcInfo;
+import org.testng.annotations.Test;
+import sun.management.GarbageCollectionNotifInfoCompositeData;
+
+import java.lang.management.GarbageCollectorMXBean;
+import java.lang.management.ManagementFactory;
+
+import static org.testng.Assert.assertFalse;
+import static org.testng.Assert.assertTrue;
+
+public class TestGarbageCollectionNotificationInfo
+{
+    @Test(expectedExceptions = NullPointerException.class, expectedExceptionsMessageRegExp = "compositeData is null")
+    public void testNullCompositeData()
+    {
+        GarbageCollectionNotificationInfo gcMonitor = new GarbageCollectionNotificationInfo(null);
+    }
+
+    @Test(timeOut = 60000)
+    public void testSimpleSuccess()
+            throws Exception
+    {
+        // First we try to force a gc
+        System.gc();
+
+        GcInfo gcInfo = null;
+        // Loop through and get a GcInfo
+        while (gcInfo == null) {
```

Comment:
In theory the spec does not guarantee the `System.gc();` to cause a full GC, so this test may (or may not) break if the java version is changed.

In practice when ` -XX:+ExplicitGCInvokesConcurrent` is set, the `System.gc();` may not trigger a full GC, effectively resulting in endless loop in this test.